[#1679] Add Tenant backend, service, and configuration

ko-j opened the pull request:
- Add additional helper classes to extract common service functions.
- Add TenantDto to wrap Tenant class and extend it with Versioned class and timestamp
- Add Tests for Tenant service
- Add dep: org.mockito:mockito-inline in order to mock final class MongoDbCallExecutor

Signed-off-by: Jan Kostulski <jan.kostulski@bosch.io>

kaniyan commented on services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/ApplicationConfig.java:
```diff
@@ -261,6 +262,17 @@ public MongoDbBasedRegistrationConfigProperties serviceProperties() {
+     * @return The properties.
+     */
+    @Bean
+    @ConfigurationProperties(prefix = "hono.tenant.mongodb")
```
This should be `@ConfigurationProperties(prefix = "hono.tenant.svc")`

ko-j replied:
resolved

kaniyan commented on services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/ApplicationConfig.java:
```diff
@@ -261,6 +262,17 @@ public MongoDbBasedRegistrationConfigProperties serviceProperties() {
+     */
+    @Bean
+    @ConfigurationProperties(prefix = "hono.tenant.mongodb")
+    public MongoDbBasedTenantsConfigProperties mongoDbTenantsProperties() {
```
IMHO we could use the same naming convention as file based device registry which is `tenantsProperties()`

ko-j replied:
resolved

ko-j committed:
[#1679] Add Tenant backend, service, and configuration

- Add additional helper classes to extract common service functions.
- Add TenantDto to wrap Tenant class and extend it with Versioned class and timestamp
- Add serviceUtils class for common functions across services
- Add ErrorHandler class for common error handling across services

Signed-off-by: Jan Kostulski <jan.kostulski@bosch.io>

ko-j committed:
[#1679] Add Tests for Tenant service
- Add dep: org.mockito:mockito-inline in order to mock final class MongoDbCallExecutor

Signed-off-by: Jan Kostulski <jan.kostulski@bosch.io>
